QRCoder/BitmapByteQRCode.cs

```diff
@@ -53,54 +55,85 @@ public byte[] GetGraphic(int pixelsPerModule, byte[] darkColorRgb, byte[] lightC
+
+        // Calculate filesize (header + pixel data + padding)
+        var fileSize = 54 + (3 * (sideLength * sideLength)) + (sideLength * paddingLen);
+
+        // Bitmap container
+        byte[] bmp = new byte[fileSize];
+        int ix = 0;
+
+        // Header part 1
+        Array.Copy(_bitmapHeaderPart1, 0, bmp, ix, _bitmapHeaderPart1.Length);
+        ix += _bitmapHeaderPart1.Length;
+
+        // Filesize
+        CopyIntAs4ByteToArray(fileSize, ix, ref bmp);
+        ix += 4;
 
-        var bmp = new List<byte>();
+        // Header part 2
+        Array.Copy(_bitmapHeaderPart2, 0, bmp, ix, _bitmapHeaderPart2.Length);
+        ix += _bitmapHeaderPart2.Length;
 
-        //header
-        bmp.AddRange(new byte[] { 0x42, 0x4D, 0x4C, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x36, 0x00, 0x00, 0x00, 0x28, 0x00, 0x00, 0x00 });
+        // Width
+        CopyIntAs4ByteToArray(sideLength, ix, ref bmp);
+        ix += 4;
+        // Height
+        CopyIntAs4ByteToArray(sideLength, ix, ref bmp);
+        ix += 4;
 
-        //width
-        bmp.AddRange(IntTo4Byte(sideLength));
-        //height
-        bmp.AddRange(IntTo4Byte(sideLength));
+        // Header end
+        Array.Copy(_bitmapHeaderPartEnd, 0, bmp, ix, _bitmapHeaderPartEnd.Length);
+        ix += _bitmapHeaderPartEnd.Length;
 
-        //header end
-        bmp.AddRange(new byte[] { 0x01, 0x00, 0x18, 0x00 });
-        bmp.AddRange(new byte[24]);
+        // Add header null-bytes
+        ix += 24;
 
-        //draw qr code
+
+        // Draw qr code
+        var group = new byte[(int)(sideLength / pixelsPerModule) * moduleDark.Length];
         for (var x = sideLength - 1; x >= 0; x -= pixelsPerModule)
         {
-            for (int pm = 0; pm < pixelsPerModule; pm++)
+            var modMatrixX = (x + pixelsPerModule) / pixelsPerModule - 1;
+
+            // Write data for first pixel of pixelsPerModule
+            int posStartFirstPx = ix;
+            for (var y = 0; y < sideLength; y += pixelsPerModule)
             {
-                for (var y = 0; y < sideLength; y += pixelsPerModule)
-                {
-                    var module =
-                        QrCodeData.ModuleMatrix[(x + pixelsPerModule) / pixelsPerModule - 1][(y + pixelsPerModule) / pixelsPerModule - 1];
-                    for (int i = 0; i < pixelsPerModule; i++)
-                    {
-                        bmp.AddRange(module ? moduleDark : moduleLight);
-                    }
-                }
-                if (sideLength % 4 != 0)
-                {
-                    for (int i = 0; i < sideLength % 4; i++)
-                    {
-                        bmp.Add(0x00);
-                    }
-                }
+                var module = QrCodeData.ModuleMatrix[modMatrixX][(y + pixelsPerModule) / pixelsPerModule - 1];
+                Array.Copy(module ? moduleDark : moduleLight, 0, group, y / pixelsPerModule * moduleDark.Length, moduleDark.Length);
```

Comment:
```suggestion
```
`group` isn't read anymore and can be removed

Reply:
Fixed in: https://github.com/codebude/QRCoder/commit/86c2e47561ef35b56f4be8f70d93c2ffd5ef40c9